Direct Debit documentation

bahmady opened the pull request:
### Context
Direct Debit documentation, to be introduced in stages for review

bahmady commented:
@katiebates 

I'm not sure of a few things:

* I've taken some liberties with edits which need reviewing: that's why I thought it would be a good idea to introduce this content in stages. Please review this as a whole: if it's too cumbersome do let me know and I'll come up with a better process.

* We differentiate between "new" users of GOV.UK Pay and existing ones, but technically someone "new" to GOV.UK Pay is still an existing user(?) Should we instead describe them as "still testing GOV.UK Pay"? This is the pattern I've experimented with in the lead-in line. 

* What does a "modified contract" mean in the context of:

> - sign a modified contract with GOV.UK Pay and our Direct Debit supplier, GoCardless 

^ Does this just mean they need to sign a new contract with Pay to reflect that they are using DD?

* Do we / will we always use GoCardless as the supplier?

* Can we say government / public sector here?

* We talk about setting up a GoCardless merchant account and how we'll provide instructions, which to me implies that we will walk them through the process of integrating DD and might weaken the documentation if I infer correctly. Are we able to word this differently or link to any GoCardless documentation?

bahmady commented:
(Everything after the section on 'agreements' at time of writing this comment):

* Did we invent the concept/name of 'agreements'? May be silly question, sorry

* I've taken away a lot of agency from "us" with these sections, i.e. "the platform" rather than "we"—is that correct or is there actually a lot of manual work on the part of GOV.UK Pay support going on behind the scenes?

* Are 'immediate' and 'delayed' payments standard industry terms? 

* This section is confusing me:

> Direct Debit payments are a 'delayed' payment method. When a user completes a
Direct Debit payment using GOV.UK Pay, the platform makes a request to their
bank to set up the Direct Debit mandate and transfer the funds. It takes
several days to confirm whether that's succeeded. 

^ I thought only the organisation being paid could set up Direct Debit. Also if it can't be confirmed, then rather than 'completed' should this be 'starts [a payment]?

bahmady commented:
@katiebates a couple of other things I'm confused about: 

* are we introducing breaking changes with the changes to the API to accommodate DD?
* do we need to tell people to use specific versions of the API to take advantage of DD?

bahmady commented:
Just want to confirm 100% that a "delayed" payment means a "delayed capture"?

katiebates commented:
> * We differentiate between "new" users of GOV.UK Pay and existing ones, but technically someone "new" to GOV.UK Pay is still an existing user(?) Should we instead describe them as "still testing GOV.UK Pay"? This is the pattern I've experimented with in the lead-in line.

We can be bolder with how we approach this documentation. It probably has more introductory stuff than required because it was written and hosted separately in Apiary.. but now it's in main docs then it doesn't need to sound like it's for a different set of users - and should flow more continuously from the rest of the docs. 

We might not need to make a distinction between new and existing users - the main differences are 
1. that users with a signed contract will have to sign an additional schedule. We don't have to call that out in docs. 
2. we want to make it clear that there's very limited additional integration required for existing users. 
3. we could be a bit less explicit about setting up merchant accounts with GoCardless - we might do more work on the OAuth flow soon to make this bit of the process easier. 

Let's discuss. 

> * What does a "modified contract" mean in the context of:
> 
> > * sign a modified contract with GOV.UK Pay and our Direct Debit supplier, GoCardless
> 
> ^ Does this just mean they need to sign a new contract with Pay to reflect that they are using DD?

I think it'll be a schedule of changes, so basically, an additional document with modifications to the original contract. 

> * Do we / will we always use GoCardless as the supplier?

Not necessarily, but most likely it'll be GC (exclusively) for the next few years. 

> * Can we say government / public sector here?

Let's discuss - not sure which bit this refers to. 

> * We talk about setting up a GoCardless merchant account and how we'll provide instructions, which to me implies that we will walk them through the process of integrating DD and might weaken the documentation if I infer correctly. Are we able to word this differently or link to any GoCardless documentation?
> * Do we expect the user to be able to figure out seamlessly how to connect a GC and Pay account?

If we improve our OAuth flow, then they won't need to set up a GC merchant account first. Again, i think we should be a bit less specific on this point until we know exactly how it's going to work. 

Let's discuss.

katiebates commented:
> @katiebates a couple of other things I'm confused about:
> 
> * are we introducing breaking changes with the changes to the API to accommodate DD?

No breaking changes. Anyone using card payments won't be affected. 

Services using DD will need to integrate with the create agreement endpoint and then make some modifications to how they call the payments endpoint to collect payments against mandates.

> * do we need to tell people to use specific versions of the API to take advantage of DD?

no - we're still on v1 even with DD 

> * just want to 100% confirm again that we're not changing e.g. rate limits, authentication processes

No we're not

bahmady commented:
Cool, thanks @katiebates —I've amended based on your feedback as much as I could think to. Let me know what you think of these sections as they stand: seems like we will have to meet to discuss in more detail but I'd like to get these early sections sorted first before adding the rest of the content, if that makes sense. If not happy to press ahead and just add them and review all in one.

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,43 @@
+[GOV.UK Pay API enables users to make
+payments](/payment_flow_overview/#payment-flow). Your contract with GOV.UK Pay
+should also cover Direct Debit, and you should have a merchant account with
+GoCardless. 
```
I think we should delete this sentence entirely (Your contract with GOV.UK Pay
should also cover Direct Debit, and you should have a merchant account with
GoCardless.) 
It doesn't explain how someone might've managed to get a contract which covers DD; also the introduction of GC is a little abrupt. 

If someone requests a DD sandbox account, the Pay team can start talking to them about the amended contract then - so we don't need to mention the contract.

bahmady replied:
OK, deleted, no sweat

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,43 @@
+* request a GOV.UK Pay Direct Debit test account [from our 
+support team](/support_contact_and_more_information/#contact-us).
+* add support for the Direct Debit endpoints to your existing GOV.UK Pay integration. 
+* sign in to your GOV.UK Pay Direct Debit account and connect your GOV.UK Pay
```
when we revisit dev on DD from Jan 2019, we might look at improving the existing OAuth flow, so instead of connecting a pre-existing GC account, you set one up via the OAuth flow initiated from Pay. So can we be a little less specific on here? Maybe something like "sign in...and use our OAuth flow to setup an account with our Direct Debit supplier." Ugh that's maybe too specific the other way... although i'm pretty sure that setting up the account via OAuth is the product direction we'll aim to go in!!

bahmady replied:
OK: is there a need to get these docs out before Jan 19? I'm happy to just put whatever you like in here but if it's going to be extensively revisited before that then perhaps we can just manage with the Apiary docs until then?

katiebates replied:
yes, we want to start showing services that we have a real DD offering. they can start integrating with us before we're ready to deliver the DD service so having API reference is useful. would prefer it to be in our tech docs rather than Apiary as it all links up better

bahmady replied:
OK, no problem

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+* payments where the amounts vary according to service usage
+* regular payments where you do not know in advance how much each payment will be
+
+You are responsible for initiating on-demand payments using the GOV.UK Pay API. 
```
Technically "initiate" is a banned word according to GOV.UK style but I actually think in the context of payments that "start" is less clear.

katiebates replied:
i'm happy for us to break the rules if you think it makes more sense!

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+
+Set up a recurring Direct Debit payment
+
+1. Create an agreement using `POST /v1/agreements/`,
```
Will we be hosting all this on Gelato before the docs go live?

katiebates replied:
i don't mind it being live before (although that would be fine), but would be great to have it in Gelato (and/or whatever we replace that with) when the docs are published.

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+6. GOV.UK Pay returns the user to your service using the `return_url` from 
+Step 1.
+
+7. Call `GET /v1/agreements/<AGREEMENT-ID>` to check the status of the
```
Is using `<AGREEMENT-ID>` like this acceptable? I'm trying to establish convention for placeholders like this across the whole docs, but want to be 100% sure that this indeed a placeholder. Previously this said: `GET /v1/agreements/:agreement_id`

katiebates replied:
not sure about the convention. the address is [GET] https://publicapi.payments.service.gov.uk/v1/agreements/[agreement_id]

bahmady replied:
I think this convention should be OK as long as the audience is devs but would be good to get confirmation from @alexbishop1 if he has a moment

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+Step 1.
+
+7. Call `GET /v1/agreements/<AGREEMENT-ID>` to check the status of the
+agreement (in order to display an appropriate screen). For a Direct Debit
```
What does "appropriate screen" mean?

katiebates replied:
it's to determine what to show the user when they return to the initiating service. for example, for successful mandates, it would be a 'your mandate was successful' screen, or if the user cancelled it, then a screen that maybe gives them another option to set up a payment.

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+payment, this will initially be in a ‘pending’ state until the bank confirms
+the mandate. The confirmation process typically takes 3-4 days.
+
+8. When the GOV.UK Pay team receives confirmation that the mandate is
```
OK to say "When the GOV.UK Pay team..." ?

katiebates replied:
hmmm probably not, suggests that we do something manual here. 'When GOV.UK Pay receives..' better.

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+
+8. When the GOV.UK Pay team receives confirmation that the mandate is
+confirmed, the GOV.UK Pay admin tool will update the transaction status. You
+can check the status of the transaction by calling the events endpoint.
```
What's the `events` endpoint, in full (and again will it be on Gelato)?

katiebates replied:
the events endpoint will show a service what's happened with a mandate or payment - we don't have something like this yet since card payments are synchronous, but DD is asynchronous so we need to give services a way to find out what's happened with a transaction instead of purely relying on gov.uk pay updating the transaction. the endpoint isn't fully developed yet. once it is, it'll go in gelato.

bahmady replied:
OK, do we want to include that in these docs?

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+* payments where the amounts vary according to service usage
+* regular payments where you do not know in advance how much each payment will be
+
+You are responsible for initiating on-demand payments using the GOV.UK Pay API. 
```
I removed this content:

> You could also use the on-demand functionality to simulate a recurring or scheduled Direct Debit payment, by sending an API call on a set schedule. Users would be emailed each time a payment is collected against their mandate.In future, we may provide functionality for scheduled, recurring Direct Debits. This is where your service provides GOV.UK Pay with a schedule for Direct Debits, and GOV.UK Pay executes that schedule with no further input from your service. Your service would specify the Direct Debit type in the agreement_type field.

Is this essential/should it go back in? Some of it feels like we cover it well enough elsewhere; other parts of it I'm not comfortable with including e.g. "In future we may ..."

katiebates replied:
i think the first two sentences are helpful - the rest can go

bahmady replied:
OK, added back

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,107 @@
+7. Call `GET /v1/agreements/<AGREEMENT-ID>` to check the status of the
+agreement (in order to display an appropriate screen). For a Direct Debit
+payment, this will initially be in a ‘pending’ state until the bank confirms
+the mandate. The confirmation process typically takes 3-4 days.
```
OK, is it worth calling out the two different situations in the docs or just saying "several days" everywhere?

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+## Refunds
+
+GOV.UK Pay does not currently allow for Direct Debit refunds. 
```
Is this correct / are we not including this?

katiebates replied:
it's correct right now and will be for DD MVP. maybe we can delete it and handle through conversations (as we'd have to explain about using the GoCardless merchant account to initiate a refund)

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+## Indemnity claims
+
+GOV.UK Pay does not report indemnity claims. You should access your GoCardless
```
Can we provide a link?

katiebates replied:
again, maybe we do the same thing as above. don't mention it but handle through conversations. the equivalent for card payments is chargebacks, and we don't call out that we don't handle them.

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+For Direct Debit transactions, GOV.UK Pay fulfils an agreement by registering
+a Direct Debit mandate with the payer’s bank. 
+
+## Synchronous and asynchronous payment methods
```
it's probably in there to explain why we've called it agreement, rather than mandate. but it's a little abstruse, if that was the intention...

bahmady replied:
I'm not sure if I follow. It might be me having made edits to the original content and forgetting, honestly (if so, apologies), or my lack of understanding, but as things stand I don't see how this reinforces "agreement" vs "mandate".

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+To set up a Direct Debit mandate to take a single payment:
+
+* create an API token linked to your Direct Debit service
```
API key. OAuth is only used to link the Pay account to the GC account on initial setup.

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+## Recurring payments using Direct Debit
+
+A recurring (or “on-demand”, or “variable”) agreement allows you to collect
```
stick with variable

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+payment service provider (PSP), but the PSP is not yet able to confirm whether
+the payment has been successful
+
+1. The agreement reaches one of several finished states:
```
yes, agreements (or mandates) have a different 'lifecycle'

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+`cancelled - user not eligible` is a new terminal state related to agreements,
+
+The agreement can go through the following state transitions.
```
agreed

bahmady commented on source/api_reference/index.html.md.erb:
```diff
@@ -164,6 +164,8 @@ Common status codes are:
 | General | P0900 | Too many requests | Your service account is sending requests above the allowed rate. Please try the request again in a few seconds. |
 | General | P0920 | Request blocked by security rules | Our firewall blocked your request. See the [“Troubleshooting”](/troubleshooting) section for details. |
 | General | P0999 | GOV.UK Pay is unavailable | The GOV.UK Pay service is temporarily down. |
+| Direct Debit | P1000 | Wrong agreement type | You can only take on-demand payments from an on-demand agreement. |
```
'General' was already in there so that might be a job for another PR / something to capture on Trello?

bahmady replied:
@jonathanglassman this might be something to pick up

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+weight: 75
+---
+
+# GOV.UK Pay Variable Direct Debit
```
Do we want to call this 'Variable Direct Debit' or just 'Direct Debit'? Either way one of the titles should change but I'm inclined to say just 'Direct Debit'? Reason being we talk about one-off payments fairly prominently.

katiebates replied:
Direct Debit is better

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+
+* request a GOV.UK Pay Direct Debit test account [from our
+support team](/support_contact_and_more_information/#contact-us).
+* add support for the Direct Debit endpoints to your existing GOV.UK Pay
```
How can users add this support?

katiebates replied:
they need to do a bit more integration work as there are new endpoints (eg to create an agreement) - probably 'support' is the wrong word..

bahmady replied:
I guess the integration tutorial will need updating as well then?

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,218 @@
+support team](/support_contact_and_more_information/#contact-us).
+* add support for the Direct Debit endpoints to your existing GOV.UK Pay
+integration
+* sign in to your GOV.UK Pay Direct Debit account and follow the instructions
```
Will this be obvious to users and will the account be fully available at the point of shipping this documentation?

katiebates replied:
they would need to find it via the 'My services' page. we know that sometimes users struggle to find this page, but we could deal with this through sending a (templated) email with steps to 1. find Pay DD account and 2. setup GC merchant account once we've set up a DD gateway account - as we'd need to notify them that it's ready.

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+* `cancelled` - Your service cancelled the payment using an API call or the self-service site
+
+* `cancelled - user not eligible` - the user indicated that they are not the sole signatory on the bank account, so the mandate setup could not be completed
```
These are not capitalised here but capitalised below. Is that correct? I know you said they were different but I just want to make 100% sure as they're worded the same.

katiebates replied:
as discussed and checked in postman - states should be capitalised

bahmady commented:
@katiebates I've resolved a few of the outstanding questions (thank you) and added some new ones for your review.

I've also made some other changes to the documentation as it stands, which is reflected in the PR now. I think it's starting to take shape. You can review it here or at:

https://govukpay-directdebit.cloudapps.digital/direct_debit

It would be especially good to get your thoughts on the new table:

https://govukpay-directdebit.cloudapps.digital/direct_debit/#agreement-lifecycle

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+## GOV.UK Pay ‘agreements’
+
+In order to support Direct Debit, GOV.UK Pay now uses ‘agreements’. An
```
slight hesitation about using 'now' as it introduces temporality - but not too worried since i'm sure we'll be updating the docs when the feature is less new

bahmady replied:
Agree, was hoping to get rid of this anyway but wanted to double-check how you wanted to pitch it

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+scheduled Direct Debit payment, by sending an API call on a set schedule.
+Users would be emailed each time a payment is collected against their mandate.
+
+You are responsible for managing variable payments using the GOV.UK Pay API.
```
i struggled a bit to understand why this final line was in there - i think it's to say that we won't take a future schedule and execute it, but i had to scratch my head a little bit to think why it's in there

bahmady replied:
Yeah I thought the same; I've deleted it.

bahmady replied:
Can always add something back in after.

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+7. Call `GET /v1/agreements/<AGREEMENT-ID>` to check the status of the
+agreement (in order to display an appropriate screen to the user). For a
+Direct Debit payment, this will initially be in a ‘pending’ state until the
+bank confirms the mandate. The confirmation process typically takes 3-4 days.
```
the mandate confirmation process takes 2 days

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+connected with. For example “testing”, “Worldpay”, or “Direct Debit”. You can
+use different API keys to direct paying users to different payment methods.
+
+### Agreement lifecycle
```
i think this needs to be an h1? or ## (not sure of the right terminology)

katiebates replied:
this section also feels a little out of place - maybe it needs to go above 'one-off payments using DD' since you'd need to choose what API key to use before embarking on one off or variable DD...

bahmady replied:
Tried moving it up there

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+## Control the payment method
+
+When you create an API key, you choose which payment gateway that key is
```
i think this should be 'gateway account' although that's the wrong language - it's about linking up the correct gov.uk Pay account with the payment method you want to use

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+1. `created` - Your service made an API call to create the agreement
+
+2. `started` - Payer has visited the `next_url` and landed on the GOV.UK Pay
+page to enter their payment details
```
might want to say bank account details here?

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+3. `pending` - Indicates that payment details have been submitted to the
+payment service provider (PSP), but the PSP is not yet able to confirm whether
+the payment has been successful
```
should be whether the mandate is successful

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+4. The agreement reaches one of several finished states:
+
+* `success` - Payment successfully completed
```
mandate completed - let's review that message from Andrew L together which contains amendments to this section

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+4. The agreement reaches one of several finished states:
+
+* `success` - Payment successfully completed
+
```
again, should relate to mandate

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+* `failed` - User attempted to make a payment but the payment did not complete
+
+* `cancelled` - Your service cancelled the payment using an API call or the self-service site
```
this should be if the user has cancelled their mandate

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+3. The payment reaches one of several finished states:
+
+  * `success` - Mandate set up successfully
```
should be about the payment

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+  * `success` - Mandate set up successfully
+
+  * `failed` - Mandate could not be set up
```
payment failed

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+
+  * `failed` - Mandate could not be set up
+
+  * `cancelled` - The user or service cancelled the mandate
```
payment was cancelled (need to doublecheck this)

bahmady replied:
Have changed mandate-> payment, will leave this comment thread in until you've double-checked

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+## Collecting payments against a mandate
+
+Once a mandate has been set up, you can make an API call to collect a payment
+against the mandate.
```
the API call can be made as soon as a mandate ID is available

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,273 @@
+exposed to the user, but will be available on GOV.UK Pay admin tool and
+through the API.
+
+Every time a payment is requested, we will send an email to the user giving
```
we should call out that a payment will take 3 days to be collected if a mandate is confirmed (+2 days to payout to the service's bank account - that could go in a different section) and that the user gets 3 days advance notice 

or payments take +4 days from the time a mandate is submitted

bahmady replied:
Had a go at adding this

katiebates commented:
> @katiebates I've resolved a few of the outstanding questions (thank you) and added some new ones for your review.
> 
> I've also made some other changes to the documentation as it stands, which is reflected in the PR now. I think it's starting to take shape. You can review it here or at:
> 
> https://govukpay-directdebit.cloudapps.digital/direct_debit
> 
@bahmady i agree this is now taking shape :) let's go through it together as there's a few edits to work out plus i think we might want to think about section ordering. thanks, looking really good!

bahmady commented:
@katiebates no problem, I've added some comments in the outstanding threads on this PR and made edits based on comments in now-resolved threads. I also changed the ordering of the information: let me know what you think. I've also updated https://govukpay-directdebit.cloudapps.digital/direct_debit

Happy to go through this whenever, might be good to put a slot in next week if you have time.

jonathanglassman commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+
+These steps happen in order.
+
+#### 1. Agreement is created
```
Is there a particular reason that these are H4s and not an ordered list?

bahmady replied:
Yeah it's less than ideal, more of a stopgap solution really. The intention with the DD docs has been to treat them like a discrete piece of work whilst development is still ongoing, rather than to integrate everything within the API reference section. Having these as a numbered list and packing too much information in messes up the Markdown, I found. There might be a clever way to do it but I ended up settling on this as a solution (for now) to make sure all the information is there and presented in an intelligible enough way

bahmady replied:
![screen shot 2019-01-10 at 12 13 15](https://user-images.githubusercontent.com/32935794/50967806-1e43f000-14d1-11e9-9102-3ea86934c3ad.png)

I was able to get this working with sub-bullets but I think it looks less clear like this

jonathanglassman replied:
fair enough

jonathanglassman commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+
+* a `return_url` where GOV.UK Pay will direct the user when they have finished setting up the agreement
+
+* the `agreement_type` (`on demand`)
```
is "on demand" an agreement type, or is there other agreement types?

bahmady replied:
Ah yes interesting point. It's synonymous with "variable" but I'm not sure why in the context of the data here it's termed "on demand". @katiebates

katiebates replied:
it's on-demand because that's what we were calling it at the time; @alexbishop1 can we refactor it to 'variable' without causing an issue?
@jonathanglassman there will (probably) be other agreement types in future - e.g. recurring/scheduled DD, so that's why we ask services to specify the type.

jonathanglassman commented on source/direct_debit/index.html.md.erb:
````diff
@@ -0,0 +1,322 @@
+```
+{
+ "amount": 2000,
+ "description": "fml",
````
Should we have a description key with the value of fml? might be interpreted as something rude

bahmady replied:
Oh my goodness

jonathanglassman commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+
+When collecting a payment against a mandate, you must supply a reference and
+description. References that your service supplies with payments will not be
+exposed to the user, but will be available on GOV.UK Pay admin tool and
```
=> "the GOV.UK Pay admin tool"

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+  <td nowrap><Something went wrong with GOV.UK Pay or the underlying Payment Service Provider></td>
+  </tr>
+</table>
+
```
Yeah it's less than ideal, more of a stopgap solution really. The intention with the DD docs has been to treat them like a discrete piece of work whilst development is still ongoing, rather than to integrate everything within the API reference section. Having these as a numbered list and packing too much information in messes up the Markdown, I found. There might be a clever way to do it but I ended up settling on this as a solution (for now) to make sure all the information is there and presented in an intelligible enough way

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+team](/support_contact_and_more_information/#contact-us).
+
+If you want to add Direct Debit to an existing integration, you should
+[contact us](/support_contact_and_more_information/#contact-us) to discuss
```
I'm not sure about this; some users are likely to be capable of doing it themselves, but there may be a middle ground. @katiebates

katiebates replied:
because the first launch will be a little rough round the edges, we think it's worth us talking through the integration details with the team integrating. they don't have to, but would be better for them if they do...

bahmady replied:
Are you happy with the language as-is in that case, @katiebates?

bahmady commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,322 @@
+  <td> Cancelled by the payer. </td>
+  </tr>
+  </table>
+
```
Ah if I understand correctly you're saying this is incorrectly indented in the .md file, right? (Which yes, I agree with). I've modified. Or do I misunderstand?

bahmady replied:
Now I understand; have created new section instead

bahmady commented:
@katiebates —

Just a small one: to be clear, the "reference number", "payment statuses", "indemnity claim" and "refund" sections apply to both one-off and variable payments?

Also, just a couple of outstanding comments @jonathanglassman has left and then if you're happy with this, we can send it off for final review and publish it.

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,324 @@
+Card payments are a synchronous payment method. When a user
+completes a card payment using GOV.UK Pay, the platform immediately confirms
+to your service whether the payment has succeeded or failed. If a user
+pays by card, you can immediately release your service to the user .
```
rogue space here

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,324 @@
+### Set up an account with the Direct Debit supplier
+
+Select your account in the __My services__ page of your GOV.UK Pay Direct
+Debit account then follow the on-screen instructions.
```
it will just be their general account rather than payment type specific, so this should say GOV.UK Pay account

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,324 @@
+    <td nowrap>Something went wrong with GOV.UK Pay or the underlying Payment Service Provider.</td>
+    </tr>
+  </table>
+</div>
```
could this table be wrapped? it's not immediately obvious that you need to scroll to see all the text

katiebates commented on source/direct_debit/index.html.md.erb:
```diff
@@ -0,0 +1,324 @@
+    <tr>
+    <td nowrap><code> STARTED </code></td>
+    <td nowrap><code> PENDING </code></td>
+    <td> Payer has provided their account details , and confirms their consent to the agreement. <code>PENDING</code> is only used with delayed payment types such as Direct Debit. This indicates that payer consent has been given but the agreement has not yet been confirmed with the issuing bank. </td>
```
rogue space

katiebates commented:
@bahmady to your first point, yes they do 


> @katiebates —
> 
> Just a small one: to be clear, the "reference number", "payment statuses", "indemnity claim" and "refund" sections apply to both one-off and variable payments?
> 
> Also, just a couple of outstanding comments @jonathanglassman has left and then if you're happy with this, we can send it off for final review and publish it.

katiebates commented:
this looks great! ready to go for writer review...

bahmady commented:
@katiebates I've wrapped the table, resolved your other comments, and added some notes about things that apply to all kinds of DD payments. Could you kindly quickly review the starting sentences of the following sections, then this is good to go:
 
"reference number", "payment statuses", "indemnity claim" and "refund"

jonathanglassman commented:
Passed tech writer 2i and ready for merge

katiebates commented:
so pleased to have this in! 

i just noticed a few things though (sorry) - i think the placement of the DD section is a bit odd since it kind of sits in the middle of card payments. Could it go after the Refunding payments section?

Also there's a sentence which says "A payment changes states as it moves through its lifecycle." - i think this is really meant to say "An agreement changes states as it moves through its lifecycle".

Sorry again, and thanks :)
